homework/shared_ptr/test.cpp

```diff
@@ -0,0 +1,222 @@
+#include <string>
+#include "gtest/gtest.h"
+#include "control_block.hpp"
+#include "shared_ptr.hpp"
+#include "weak_ptr.hpp"
+
+const std::string testString{"Ala ma kota"};
+constexpr int testValueOne = 10;
+constexpr int testValueTwo = 20;
+const std::string deleterString = "Deleted";
+constexpr int testNumber = 0;
+
+struct A {
+    int a;
+    std::string b;
+};
+
+std::function<void(A* ptr)> deleter = [](A* ptr) { ptr->a = 0; ptr->b = "Deleted"; };
+
+struct sharedPtrTest : ::testing::Test {
+    sharedPtrTest()
+    : sPtr(new int{testValueOne}) {}
+    cs::shared_ptr<int> sPtr;
+};
+
+TEST_F(sharedPtrTest, testCounter) {
+    ASSERT_EQ(sPtr.use_count(), 1);
+    {
+        cs::shared_ptr<int> testPtr(sPtr);
+        ASSERT_EQ(sPtr.use_count(), 2);
+    };
+    ASSERT_EQ(sPtr.use_count(), 1);
+}
+
+TEST_F(sharedPtrTest, testCopyConstructor) {
+    ASSERT_EQ(*sPtr, testValueOne);
+    ASSERT_EQ(sPtr.use_count(), 1);
+    auto sPtr2(sPtr);
+    auto sPtr3 = sPtr;
+    ASSERT_EQ(*sPtr, *sPtr2);
+    ASSERT_EQ(*sPtr, *sPtr3);
+    ASSERT_EQ(sPtr3.use_count(), 3);
+}
+
+TEST_F(sharedPtrTest, testCopyAssignment) {
+    ASSERT_EQ(*sPtr, testValueOne);
+    ASSERT_EQ(sPtr.use_count(), 1);
+    cs::shared_ptr<int> sPtr2;
```

Comment:
```suggestion
    cs::shared_ptr<int> sPtr2{new int{100}};
```
This may cause some memory leaks if you have bad implementation, so it is better to use it in tests. The same in other places and tests like testMoveAssignment.

Reply:
done
we ave good implementation - no memory leaks here :)